### PR TITLE
Logs: Add tracking for permalinking

### PR DESCRIPTION
**Pull request by svennergr**
**What is this feature?**

This PR adds feature tracking for log line permalinking.

### CODE REVIEW

File: public/app/features/explore/Logs/Logs.test.tsx [hunk 1]
```diff
@@ -243,6 +271,49 @@ describe('Logs', () => {
 
       expect(scrollElementMock.scroll).not.toHaveBeenCalled();
     });
+
+    it('should call reportInteraction on permalinkClick', async () => {
+      Object.defineProperty(window, 'location', {
```
Comment (matyax) [inline public/app/features/explore/Logs/Logs.test.tsx:276]: I might be wrong in 2023, but I think you need to restore these modifications in an `afterAll`.

File: public/app/features/explore/Logs/Logs.tsx [hunk 2]
```diff
@@ -368,6 +368,11 @@ class UnthemedLogs extends PureComponent<Props, State> {
     const baseUrl = /.*(?=\/explore)/.exec(`${window.location.href}`)![0];
     const url = urlUtil.renderUrl(`${baseUrl}/explore`, { left: serializedState });
     await createAndCopyShortLink(url);
+
+    reportInteraction('grafana_explore_logs_permalink_clicked', {
+      datasourceType: row.datasourceType ?? 'unknown',
+      logRowUid: row.uid,
```
Comment (matyax) [inline public/app/features/explore/Logs/Logs.tsx:374]: Just brainstorming. Might it be useful to collect something about the logged line? Maybe level? (e.g. are they sharing more errors, or warnings, or just any log?).

**Review by matyax — COMMENTED** [review]
Looks good! Left a few suggestions.

File: public/app/features/explore/Logs/Logs.tsx (hunk 2 above)
Comment (svennergr) [reply public/app/features/explore/Logs/Logs.tsx:374]: Good idea! Added in f0131da26ee2286145c182d54321f111e8825090.

**Review by matyax — APPROVED** [review]
☄️

**Pull request merged**